src/bootloader.rs

```diff
@@ -157,14 +158,29 @@ impl<ObjectID: FsVerityHashValue> Type1Entry<ObjectID> {
                 let file = self.files.remove(value);
 
                 let new = format!("/{entry_id}/{basename}");
+                let new_entry_path = Path::new(&new);
```

Comment:
Except yeah, above we have a string and so...this is where camino makes sense. It's honestly also one of the reasons I just ended up strictly denying non-UTF8 filenames in ostree, dealing with it is just a giant pain and there's so many file formats which reference filenames that just don't support non-UTF8 filenames, or would be extremely awkward to do.